File: benchmarks/fibonacci/fibonacci.zig
const std = @import("std");

// Configuration
const N1 = 10;
const N2 = 20;
const N3 = 30;
const N4 = 35;

// Markers matching Lamina version
const HEADER_MARKER: i64 = 123456789;
const FOOTER_MARKER: i64 = 987654321;

fn fibonacciIterative(n: u32) i64 {
    if (n == 0) return 0;
    if (n == 1) return 1;

    var a: i64 = 0;
    var b: i64 = 1;
    var i: u32 = 2;
    while (i <= n) : (i += 1) {
        const temp = a + b;
        a = b;
        b = temp;
    }
    return b;
}

pub fn main() void {
    // Print header marker
    std.debug.print("{d}\n", .{HEADER_MARKER});

    // Compute and print fibonacci numbers
    std.debug.print("{d}\n", .{fibonacciIterative(N1)});
    std.debug.print("{d}\n", .{fibonacciIterative(N2)});
    std.debug.print("{d}\n", .{fibonacciIterative(N3)});
    std.debug.print("{d}\n", .{fibonacciIterative(N4)});

    // Print footer marker
    std.debug.print("{d}\n", .{FOOTER_MARKER});
}

File: benchmarks/fibonacci/fibonacci.rb
#!/usr/bin/env ruby

# Configuration
N1 = 10
N2 = 20
N3 = 30
N4 = 35

# Markers matching Lamina version
HEADER_MARKER = 123456789
FOOTER_MARKER = 987654321

def fibonacci_iterative(n)
  if n == 0
    return 0
  end
  if n == 1
    return 1
  end

  a = 0
  b = 1
  (2..n).each do |i|
    temp = a + b
    a = b
    b = temp
  end
  return b
end

# Print header marker
puts HEADER_MARKER

# Compute and print fibonacci numbers
puts fibonacci_iterative(N1)
puts fibonacci_iterative(N2)
puts fibonacci_iterative(N3)
puts fibonacci_iterative(N4)

# Print footer marker
puts FOOTER_MARKER

